api/platform/v1alpha1/redeployment_types.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// the types of Kubernetes workloads
+// +kubebuilder:validation:Enum=daemonsets;deployments;statefulsets
+type WorkflowType string
```

Comment:
Can we define already the initially supported types as consts?

Ref:
https://github.com/kubernetes/kubernetes/blob/58c10aa6eb5adfb1f3aa4d6cb898b8c347ba9e72/staging/src/k8s.io/api/core/v1/types.go#L2618-L2641

Reply:
yes, it will be useful for the controller. I will add this